tools/modules/update_payload_cached_sizes.rb

```diff
@@ -31,17 +31,13 @@
     next if name =~ /generic/
     mod_inst = framework.payloads.create(name)
     #mod_inst.datastore.merge!(framework.datastore)
-    next if Msf::Util::PayloadCachedSize.is_cached_size_accurate?(mod_inst)
+    next if mod_inst.is_a?(Msf::Payload::Adapter) || Msf::Util::PayloadCachedSize.is_cached_size_accurate?(mod_inst)
     $stdout.puts "[*] Updating the CacheSize for #{mod.file_path}..."
     Msf::Util::PayloadCachedSize.update_module_cached_size(mod_inst)
   rescue => e
+    $stderr.puts "[!] Caught Error while updating #{name}:\n#{e}\n#{e.backtrace.map { |line| "\t#{line}" }.join("\n")}"
     exceptions << [ e, name ]
-    next
   end
 end
 
-exceptions.each do |e, name|
-  print_error("Caught Error while updating #{name}:\n#{e}")
```

Comment:
`print_error` here is not framework's DSL - it's actually a random helper that's accidentally being loaded:

https://github.com/rapid7/metasploit-framework/blob/772cf99d368882bec24bc33763bb86f4737fa70b/lib/msf/core/web_services/http_db_manager_service.rb#L79-L103

This should be fixed - but not in this PR

Reply:
The errors were all printed at the end to ensure they were not lost in noise if a large number of payloads changed size, should this we keep that pattern?

``` ruby
exceptions.each do |e, name|
  $stderr.puts "[!] Caught Error while updating #{name}:\n#{e}\n#{e.backtrace.map { |line| "\t#{line}" }.join("\n")}"
end
```

Reply:
> should this we keep that pattern?

I'd prefer not to keep that pattern; I think the 'print errors at the end' workflow was just a work around for the underlying errors not being fixed. Printing inline errors makes the most sense to be so the issues can be identified and fixed sooner